### ISSUE TITLE
Exception error

### ISSUE THREAD

**Issue by lljohnston** [utlib/BatchUpload#6]
I'm working with a group running 2.6.1.

I went to test BatchUpload for the first time and got this exception error:

RuntimeException: The configured PHP path (/usr/bin/php) is invalid. in /home/icgarchives/public_html/omeka-2.6.1/application/libraries/Omeka/Job/Process/Dispatcher.php:114
Stack trace:
#0 /home/icgarchives/public_html/omeka-2.6.1/application/libraries/Omeka/Job/Process/Dispatcher.php(93): Omeka_Job_Process_Dispatcher::_checkCliPath('/usr/bin/php')
#1 /home/icgarchives/public_html/omeka-2.6.1/application/libraries/Omeka/Job/Process/Dispatcher.php(28): Omeka_Job_Process_Dispatcher::getPHPCliPath()
#2 /home/icgarchives/public_html/omeka-2.6.1/application/libraries/Omeka/Job/Dispatcher/Adapter/BackgroundProcess.php(31): Omeka_Job_Process_Dispatcher::startProcess('Omeka_Job_Proce...', Object(User), Array)
#3 /home/icgarchives/public_html/omeka-2.6.1/application/libraries/Omeka/Job/Dispatcher/Default.php(151): Omeka_Job_Dispatcher_Adapter_BackgroundProcess->send('{"className":"B...', Array)
#4 /home/icgarchives/public_html/omeka-2.6.1/plugins/BatchUpload/libraries/BatchUpload/Wizard/ExistingCollection.php(151): Omeka_Job_Dispatcher_Default->sendLongRunning('BatchUpload_Job...', Array)
#5 /home/icgarchives/public_html/omeka-2.6.1/plugins/BatchUpload/libraries/BatchUpload/Application/AbstractWizard.php(84): BatchUpload_Wizard_ExistingCollection->step2Process(Array)
#6 /home/icgarchives/public_html/omeka-2.6.1/application/libraries/Omeka/Plugin/Broker.php(154): BatchUpload_Application_AbstractWizard->_stepProcess(Array)
#7 /home/icgarchives/public_html/omeka-2.6.1/application/libraries/globals.php(188): Omeka_Plugin_Broker->callHook('batch_upload_ex...', Array)
#8 /home/icgarchives/public_html/omeka-2.6.1/plugins/BatchUpload/controllers/JobsController.php(155): fire_plugin_hook('batch_upload_ex...', Array)
#9 /home/icgarchives/public_html/omeka-2.6.1/application/libraries/Zend/Controller/Action.php(516): BatchUpload_JobsController->wizardAction()
#10 /home/icgarchives/public_html/omeka-2.6.1/application/libraries/Zend/Controller/Dispatcher/Standard.php(308): Zend_Controller_Action->dispatch('wizardAction')
#11 /home/icgarchives/public_html/omeka-2.6.1/application/libraries/Zend/Controller/Front.php(954): Zend_Controller_Dispatcher_Standard->dispatch(Object(Zend_Controller_Request_Http), Object(Zend_Controller_Response_Http))
#12 /home/icgarchives/public_html/omeka-2.6.1/application/libraries/Zend/Application/Bootstrap/Bootstrap.php(105): Zend_Controller_Front->dispatch()
#13 /home/icgarchives/public_html/omeka-2.6.1/application/libraries/Zend/Application.php(384): Zend_Application_Bootstrap_Bootstrap->run()
#14 /home/icgarchives/public_html/omeka-2.6.1/application/libraries/Omeka/Application.php(73): Zend_Application->run()
#15 /home/icgarchives/public_html/omeka-2.6.1/admin/index.php(28): Omeka_Application->run()
#16 {main}

**Comment by lljohnston** [comment]
I also got this error:

Deprecated: The each() function is deprecated. This message will be suppressed on further calls in /home/icgarchives/public_html/omeka-2.6.1/application/libraries/Zend/Cache/Backend.php on line 79

**Comment by dicksonlaw583** [comment]
Your first error is caused by an incorrectly configured Omeka job queue. If you are running Omeka on your own system, you can run `which php`, then open the Omeka instance's `application/config/config.ini` file and set its `background.php.path` to the output of `which php`. If you are running Omeka on a web hosting service, please ask your provider for the correct value. To verify your setup, you can use [this plugin](https://github.com/utlib/JobDiagnostics).

Your second error can be fixed by upgrading to [Omeka 2.7](https://github.com/omeka/Omeka/releases/tag/v2.7).

**Comment by lljohnston** [comment]
Upgraded to 2.7 and the second issue was fixed, thanks.

I'm having a weird issue. I've updated the config.ini and I'm still getting the runtime exception, but it's like it's not reading the config.ini file because the error shows a value that is NOT what's specified in the ini file.

I've got 

background.php.path = "/usr/local/bin/php"

in the ini but getting the error

RuntimeException: The configured PHP path (/usr/bin/php) is invalid. in /home/icgarchives/public_html/omeka-2.7/application/libraries/Omeka/Job/Process/Dispatcher.php:114

Any insight into this?

running which php definitely shows /usr/local/bin/php as the path

The plugin is running fine on my other 2.7 instance with the config.ini change, but that's on Reclaim.

**Comment by dicksonlaw583** [comment]
What happens when you run this?
```
/usr/local/bin/php -v
```
Omeka's dispatcher expects something looking like this in the first line of the output, and if not found it will throw the exception you described:
```
PHP 7.2.22
```
Source: https://github.com/omeka/Omeka/blob/v2.7/application/libraries/Omeka/Job/Process/Dispatcher.php#L102

**Comment by lljohnston** [comment]
This is what the volunteer sys admin says:

> In answer to his question, here is the first line from running "/usr/local/bin/php -v":
> 
> PHP 7.2.24 (cli) (built: Oct 30 2019 15:00:52) ( NTS )
> 
> The point is that it doesn't seem to get that far. It looks like escapeshellarg to get the value for $cliPath returns nothing so the command being exec-d is just " -v".  I still haven't gotten time to run a stand-alone unit test on this function.

**Comment by dicksonlaw583** [comment]
That should do fine. In that case, please double-check to see that it's `config.ini` being changed, not `config.ini.changeme`, and that the edit has gone through. You can also try overwriting this line in your installation manually: https://github.com/omeka/Omeka/blob/v2.7/application/libraries/Omeka/Job/Process/Dispatcher.php#L28

Please also try asking around on the Omeka forum for information on the `background.php.path` problem, as the issue is somewhat common and not specific to this plugin.

**Comment by lljohnston** [comment]
Overwriting it with what? This?

$cliPath = /usr/bin/local/php;

**Comment by dicksonlaw583** [comment]
```
$cliPath = "/usr/bin/local/php";
```

**Comment by lljohnston** [comment]
I get this error:

Warning: escapeshellarg() has been disabled for security reasons in /home/icgarchives/public_html/omeka-2.7/application/libraries/Omeka/Job/Process/Dispatcher.php on line 48

Warning: escapeshellarg() has been disabled for security reasons in /home/icgarchives/public_html/omeka-2.7/application/libraries/Omeka/Job/Process/Dispatcher.php on line 49

Warning: escapeshellarg() has been disabled for security reasons in /home/icgarchives/public_html/omeka-2.7/application/libraries/Omeka/Job/Process/Dispatcher.php on line 50

**Comment by dicksonlaw583** [comment]
See this fix: https://stackoverflow.com/a/32702053

Please consult with your system administrator regarding these error messages, as they are not tied specifically to this plugin.

**Comment by dicksonlaw583** [comment]
Fixed in 1.1.0.
